Implementation of new form related designs

oguzeroglu opened the pull request:
Closes #1708

oguzeroglu commented:
@nazlikaya It is ready for a visual test. Please note that this pull request only covers the visual aspects but not the form validation logic so do your tests accordingly. We'll cover the deployment config modal validation logic related work in a separate PR.

Also make sure not to approve the pull request via the Github interface as we don't want to merge this PR yet, there needs to be a code review too.

nazlikaya commented:
Design Review
- When the field has focus, it has 2 blue colors.
- When the field in error case, then the red focus style also has 2 red colors.
- Horizontal alignment : Please check the left hand side alignment for all the input fields. They need to be aligned according to "Deploy Diagram" header
- Vertical alignment : The starting point of the input fields right now too far away from the dialog header, please adjust it according to the design.
- Tenant id field needs to have a hint as "Optional".
- Text field width -> 556px , dialog width -> 636px

oguzeroglu commented:
@nikku I refactored the code in a way that we produce Radio/Text/Password/Checkbox etc. fields in the same way as Bootstrap and these classes are all globally implemented.

For instance what we produce for Radio:

![Screenshot 2020-03-05 at 20 17 44](https://user-images.githubusercontent.com/15003836/76017149-64061800-5f1e-11ea-9fc3-e1ce442f0dbe.png)

The code is ready for a review, however I'll keep the [WIP] tag as we don't want to merge this before Nazli approves.

@nazlikaya I fixed the visual bugs. Could you check again please? (includibg Start Instance and Privacy Prefs. modals)

nikku commented on client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceConfigModal.less:
```diff
@@ -7,74 +7,30 @@
+  fieldset {
+    border-style: none;
+    margin: 20px auto;
+    padding-left: 0 !important;
```
Let's not use !important here.

oguzeroglu replied:
👍

nikku commented on client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceConfigModal.less:
```diff
@@ -7,74 +7,30 @@
+    padding-right: 0 !important;
   }
 
   .fields {
```
This is shared across dialogs and should be global I assume?

nikku commented on client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceConfigModal.less:
```diff
@@ -7,74 +7,30 @@
-    padding: 15px;
-  }
-
   legend {
```
This is shared across dialogs and should be global I assume?

nikku commented on client/src/plugins/camunda-plugin/deployment-tool/DeploymentConfigModal.js:
```diff
@@ -208,54 +184,40 @@ export default class DeploymentConfigModal extends React.PureComponent {
+                    </p>
+                  )
+                }
+                <fieldset className='first-fieldset'>
```
Let's not use `first-child` / `last-child` marker classes.

In CSS there is better semantic support for that, either via `:first-child` or via `.intro + fieldset` selectors.

nikku replied:
Even better, what you'd generally do is define the distance between following fieldsets:

```css
fieldset + fieldset {
   margin-bottom: 30px;
}
```

oguzeroglu replied:
I squashed the first-child/last-child solution into e756492fffab26f12a9e8d892cf8a3df9aec30b9

nikku commented on client/src/styles/_modal.less:
```diff
@@ -24,7 +24,7 @@
   right: 0;
 
-  width: 636px;
+  width: 638px;
```
I like that change. :mag:

oguzeroglu replied:
Due to bordering this was the only way to give exact 636 pixels of width to modals actually. With 636 pixels assigned, the modals would have 634 pixels of width. Took me a while to fix this.

oguzeroglu replied:
<img width="843" alt="Screenshot 2020-03-06 at 10 51 19" src="https://user-images.githubusercontent.com/15003836/76072540-71f87f00-5f98-11ea-9338-a1c48520f59a.png">

nikku replied:
I personally think we should bot be picky about two pixels. 

Otherwise we'd need to specify what the width is about => Includes border? Does not include border?

oguzeroglu replied:
To me it's no big deal neither. It's just that the design review includes the exact with of the modal as well as the exact width of the input fields, that's why I modified here, to make sure the PR passes the design review.

nikku replied:
What was specified? Should the width include the border or not?

oguzeroglu replied:
I'm  not exactly sure, this is how it looks like in the Zeplin:
<img width="774" alt="Screenshot 2020-03-06 at 11 45 30" src="https://user-images.githubusercontent.com/15003836/76076954-20ec8900-5fa0-11ea-8ae3-6105471f239b.png">

@nazlikaya Could you clarify this conversation?

nikku replied:
Yea. Just checked, Zeplin, too. I guess it shows the outer width of elements (including potential border).

@nazlikaya So given we had a 5px border, an element in Zeplin would still have the same width it had with a 1px border?

nazlikaya replied:
The number Zeplin shows the total width and height of the element (including borders).

oguzeroglu replied:
Ok, I take this modification back.

oguzeroglu replied:
Update: Nazli says we should have exactly 636 px's visible in Chrome Dev Tool so I'm keeping this field as 638px. Had to update Checkbox and Radio widths as well.

oguzeroglu commented:
@nikku the commit 566b3b6 fixes the Token label floating issue 🥇

nikku commented:
Isn't `display: block` for the `label` the right fix for the issue? What https://github.com/camunda/camunda-modeler/commit/566b3b69c63d074be02989c27a440802bafe667b proposes looks like a hack to me.

oguzeroglu commented:
Yeah, what you propose is a better solution it seems. I squashed it into the first commit.

oguzeroglu commented:
Globalization of fields, fieldset, legend, form font size related styles are implemented via 808da07

nazlikaya commented:
Everything looks great! Thanks @oguzeroglu

oguzeroglu commented:
@nazlikaya Thanks!

@nikku I think we can merge this branch!
